doc/source/tutorial/stats/sampling_dgt.rst

```diff
@@ -0,0 +1,127 @@
+of the given probability vector can be set by the `guide_factor` parameter:
+
+    >>> import numpy as np
+    >>> from scipy.stats import DiscreteGuideTable
+    >>>
+    >>> pv = [0.18, 0.02, 0.8]
+    >>> urng = np.random.default_rng()
+    >>> rng = DiscreteGuideTable(pv, random_state=urng)
+    >>> rng.rvs()
+    2
+
+By default, the probability vector is indexed starting at 0. However, this
+can be changed by passing a ``domain`` parameter. When ``domain`` is given
+in combination with the PV, it has the effect of relocating the
+distribution from ``(0, len(pv))`` to ``(domain[0], domain[0] + len(pv))``.
+``domain[1]`` is ignored in this case.
+
+    >>> rng = DiscreteGuideTable(pv, random_state=urng, domain=(10, 13))
+    >>> rng.rvs()
+    10
+
+The method also works when no probability vector but a PMF is given.
+In that case, a bounded (finite) domain must also be given either by
+passing the ``domain`` parameter explicitly or by providing a ``support``
+method in the distribution object:
+
+    >>> class Distribution:
+    ...     def __init__(self, c):
+    ...             self.c = c
+    ...     def pmf(self, x):
+    ...             return x ** self.c
+    ...     def support(self):
+    ...             return 0, 10
+    ...
+    >>> dist = Distribution(2)
+    >>> rng = DiscreteGuideTable(dist, random_state=urng)
+    >>> rng.rvs()
+    9
+
+.. note:: As :class:`~DiscreteGuideTable` expects PMF with signature
+          ``def pmf(self, x: float) -> float``, it first vectorizes the
+          PMF using ``np.vectorize`` and then evaluates it over all the
+          points in the domain. But if the PMF is already vectorized,
+          it is much faster to just evaluate it at each point in the domain
+          and pass the obtained PV instead along with the domain.
+          For example, ``pmf`` methods of SciPy's discrete distributions
+          are vectorized and a PV can be obtained by doing:
+
+          >>> from scipy.stats import binom, DiscreteGuideTable
+          >>> dist = binom(10, 0.2)  # distribution object
+          >>> domain = dist.support()  # the domain of your distribution
+          >>> x = np.arange(domain[0], domain[1] + 1)
+          >>> pv = dist.pmf(x)
+          >>> rng = DiscreteGuideTable(pv, domain=domain)
+
+          Domain is required here to relocate the distribution
+
+The size of the guide table relative to the probability vector may be set using
+the ``guide_factor`` parameter. Larger guide tables result in faster generation
+time but require a more expensive setup.
+
+    >>> guide_factor = 2
+    >>> rng = DiscreteGuideTable(pv, random_state=urng, guide_factor=guide_factor)
+    >>> rng.rvs()
+    2
+
+Unfortunately, the PPF is rarely available in closed form or too slow when
+available. For many distributions, the CDF is also not easy to obtain. This
+method addresses both the shortcomings. The user only has to provide the
```

Comment:
This method doesn't address both shortcomings i.e. only provides `ppf` and not `cdf`. I think we should remove the sentence about CDF.

Reply:
```suggestion
available. The user only has to provide the
```